### PR TITLE
Ygo 37

### PR DESCRIPTION
**Pull request by NightAngell**
There are no tests for DecklistService.FindAll because is too complex to test it now. Also there is no tests for
DateValidator.IsValidFormat because regex inside that class was tested manually by me using https://regex101.com/ .
image
image
image

### CODE REVIEW

**Comment by NightAngell** [comment]
Ps. I tested DecklistService.FindAll manually :)

File: YGOProAnalyticsServer/Controllers/DecklistController.cs [hunk 1]
```diff
@@ -19,15 +22,62 @@ public class DecklistController : ControllerBase
...
+            if (!_decklistBrowserQueryParamsValidator.IsValid(queryParams))
+            {
+                return BadRequest("Invalid params. Please report it to administrator.");
+            }
+
+            DateTime? statisticsFrom = null;
+            DateTime? statisticsTo = null;
+            if (!string.IsNullOrEmpty(queryParams.StatisticsFromDate))
+            {
+                statisticsFrom = DateTime.Parse(queryParams.StatisticsFromDate);
```
Comment (szefo09) [inline YGOProAnalyticsServer/Controllers/DecklistController.cs]: DateTime Parse may be an issue if the front and backend use different DateTime models

File: YGOProAnalyticsServer/Controllers/DecklistController.cs [hunk 2]
```diff
@@ -19,15 +22,62 @@ public class DecklistController : ControllerBase
...
+            }
+
+            if (!string.IsNullOrEmpty(queryParams.StatisticsToDate))
+            {
+                statisticsTo = DateTime.Parse(queryParams.StatisticsToDate);
```
Comment (szefo09) [inline YGOProAnalyticsServer/Controllers/DecklistController.cs]: DateTime.Parse may be an issue

File: YGOProAnalyticsServer/Services/Others/BanlistService.cs [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+{
+    /// <summary>
+    /// Provide features related with banlists.
+    /// </summary>
+    /// <seealso cref="YGOProAnalyticsServer.Services.Others.Interfaces.IBanlistService" />
+    public class BanlistService : IBanlistService
+    {
+        readonly YgoProAnalyticsDatabase _db;
+
+        public BanlistService(YgoProAnalyticsDatabase db)
```
Comment (szefo09) [inline YGOProAnalyticsServer/Services/Others/BanlistService.cs:25]: Create comments for the constructor

File: YGOProAnalyticsServer/Services/Others/BanlistService.cs [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+                }
+            }
+        }
+
+        /// <summary>
+        /// CardId with info about how many copies this card is in deck.
+        /// </summary>
+        public class CardWithInfoAboutNumberOfCopiesInDeck
+        {
+            public CardWithInfoAboutNumberOfCopiesInDeck(int cardId)
```
Comment (szefo09) [inline YGOProAnalyticsServer/Services/Others/BanlistService.cs:134]: Constructor is missing comments

File: YGOProAnalyticsServer/Services/Others/DecklistService.cs [hunk 5]
```diff
@@ -1,35 +1,266 @@
...
+    /// </summary>
     public class DecklistService : IDecklistService
     {
-        YgoProAnalyticsDatabase _db;
+        readonly YgoProAnalyticsDatabase _db;
+        readonly IBanlistService _banlistService;
+        readonly IMemoryCache _cache;
 
-        public DecklistService(YgoProAnalyticsDatabase db)
+        public DecklistService(
```
Comment (szefo09) [inline YGOProAnalyticsServer/Services/Others/DecklistService.cs:37]: Constructor is missing comments

**Review by szefo09 — APPROVED** [review]

**Pull request merged**